pkg/agent/memberlist/cluster.go

```diff
@@ -422,6 +424,16 @@ func (c *Cluster) syncConsistentHash(eipName string) error {
 		klog.V(4).InfoS("Finished syncing consistentHash", "ExternalIPPool", eipName, "durationTime", time.Since(startTime))
 	}()
 
+	if eipName == allNodesConsistentHashMapKey {
+		allNodes := c.AliveNodes()
+		c.consistentHashRWMutex.Lock()
+		defer c.consistentHashRWMutex.Unlock()
+		allNodesConsistentHashMap := NewNodeConsistentHashMap()
+		allNodesConsistentHashMap.Add(allNodes.UnsortedList()...)
```

Comment:
It should only add the intersection of aliveNodes and nodes got from nodeLister, in case of stale agents running on deleted nodes.